cmake/modules/doc.cmake

```diff
@@ -0,0 +1,22 @@
+# SPDX-License-Identifier: Apache-2.0
+#
+# Copyright (c) 2021, Nordic Semiconductor ASA
+
+# This CMake module will load all Zephyr CMake modules required for a
+# documentation build.
+#
+# The following CMake modules will be loaded:
+# - extensions
+# - python
+# - west
+# - root
+# - zephyr_module
+#
+# Outcome:
+# The Zephyr package required for documentation build setup.
+
```

Comment:
strictly speaking we don't, cause all the modules below has it.

And as long as this code doesn't setup variables / define functions and similar which must not be re-loaded then a include guard is not required.

But to be on safe side in case there are changes to this code in future then I think it's nice to add a guard, so that has been done.